Hyper/PlayerData/InventorySystem/Items/Pickaxes/Pickaxe.cs

```diff
@@ -63,20 +51,32 @@ private void ModifyTerrain(Scene scene, IChunkWorker chunkWorker, float time, Fu
                 otherSphereLocation = GetOtherSphereLocation(scene.Camera.Sphere, location, scene);
             }
 
+            ModificationArgs modificationArgs = new ModificationArgs();
+
             foreach (var chunk in chunkWorker.Chunks)
             {
                 if (chunk.DistanceFromChunk(location) < Radius)
                 {
-                    modifier(chunk)(location, time + modificationTime, BrushWeight, Radius); // TODO why this is not done on the other thread? *Possible* advantages: faster & eliminate race condition we potentially get in ChunkWorker, line 179
-                    chunkWorker.EnqueueUpdatingChunk(chunk);
+                    modificationArgs.ModificationType = modificationType;
+                    modificationArgs.Location = location;
+                    modificationArgs.Time = time + modificationTime;
+                    modificationArgs.BrushWeight = BrushWeight;
+                    modificationArgs.Radius = Radius;
+                    modificationArgs.Chunk = chunk;
+                    chunkWorker.EnqueueModification(modificationArgs);
                 }
 
                 if (otherSphereLocation == null)
                     continue;
                 if (chunk.DistanceFromChunk(otherSphereLocation.Value) < Radius)
                 {
-                    modifier(chunk)(otherSphereLocation.Value, time + modificationTime, BrushWeight, Radius);
-                    chunkWorker.EnqueueUpdatingChunk(chunk);
+                    modificationArgs.ModificationType = modificationType;
```

Comment:
should these be the same?

Reply:
if so we can just reuse the whole args. If not I think we need to create completely new args

Reply:
I can move setting some of these fields before the for loop, but chunks fields needs to be set here. I don't think we need to create new args, it's passed by value anyway

Reply:
Ok so I think we should set these before the if. However I still dont understand what happens here. Am i blind or are both the calls exactly the same

Reply:
I think the issue is that the location is wrong. We used to pass otherSphereLocation.Value

Reply:
Btw it does not work in spherical.